File: scripts/convert_to_h5ad.py
# convert_to_h5ad.py
import sys
import scipy.io
import scanpy as sc
import pandas as pd

# Arguments
input_dir = sys.argv[1]

# File paths
counts_path = f'{input_dir}/counts.mtx'
metadata_path = f'{input_dir}/metadata.csv'
gene_names_path = f'{input_dir}/gene_names.csv'

# Read the MatrixMarket file in Python
X = scipy.io.mmread(counts_path)

# Create AnnData object
adata = sc.AnnData(X=X.transpose().tocsr())

# Load cell metadata
cell_meta = pd.read_csv(metadata_path)

# Load gene names
with open(gene_names_path, 'r') as f:
    gene_names = f.read().splitlines()

# Assign metadata and gene names to AnnData
adata.obs = cell_meta
adata.var.index = gene_names
adata.obs_names = adata.obs["cell_id"] 

output_file = f'{input_dir}/adata.h5ad'

# Save the AnnData object as h5ad
adata.write(output_file)

File: scripts/convert_to_h5ad.R
# convert_to_h5ad.R
args <- commandArgs(trailingOnly = TRUE)

# Arguments
seuratObj_path <- args[1]
output_dir <- args[2]

# Load libraries
library(Seurat)
library(Matrix)

# Load data
load(seuratObj_path) # Modify according to your data format

# Create the directory if it doesn't exist
if (!dir.exists(output_dir)) {
  dir.create(output_dir, recursive = TRUE)
}
seuratObj@meta.data$cell_id <- rownames(seuratObj@meta.data)

# Set the working directory to the output directory
setwd(output_dir)

# Create Seurat object
# Write metadata to CSV
write.csv(seuratObj@meta.data, file=paste0(output_dir, "/metadata.csv"), quote=FALSE, row.names=FALSE)

# Get counts matrix and write to MatrixMarket format
counts_matrix <- GetAssayData(seuratObj, assay='RNA', layer='counts')
writeMM(counts_matrix, file=paste0(output_dir, "/counts.mtx"))

# Write gene names to CSV
write.table(data.frame('gene' = rownames(counts_matrix)), file=paste0(output_dir, "/gene_names.csv"), quote=FALSE, row.names=FALSE, col.names=FALSE)
